activerecord/test/cases/migration/pending_migrations_test.rb

```diff
@@ -100,6 +115,45 @@ def test_understands_migrations_created_out_of_order
             check_pending.call({})
           end
         end
+
+        def test_with_multiple_database
```

Comment:
@eileencodes I've updated the test case here to better reflect what the code is trying to do. If the connection pool for a database with a pending migration has not been created, the pending migration error is not thrown. Instead, we only check for pending migrations when a new connection pool that hasn't been seen by the middleware is detected during the request.